missions/W5/M2/nyc_taxi_analysis.py

```
@@ -0,0 +1,78 @@
+    arg_category = sys.argv[3]
+    arg_path = sys.argv[4]
+    print(f"Year: {arg_year}, Month: {arg_month}, Category: {arg_category}, Path: {arg_path}")
+    spark = SparkSession.builder.appName("NYC Taxi Analysis").getOrCreate()
+    spark.sparkContext.setLogLevel("ERROR")
+    df = extract_data_df(arg_year, arg_month, arg_category, arg_path, spark)
+    df = filter_data(df, arg_year, arg_month)
+    df = add_columns(df)
+    df.cache()
+    hourly_df = group_by_hourly_data(df)
+    hourly_df.cache()
+    hourly_weather_df = spark.read.csv(arg_path+f"nyc_weather_data/nyc_taxi_weather_hourly_data.csv")
```

Ruff (0.8.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)